ui/v2.5/src/components/Shared/StashBoxIDSearchModal.tsx

```diff
@@ -0,0 +1,445 @@
+  onSelectItem: (stashId: string, endpoint: string) => void;
+}
+
+// Shared component for rendering images
+const SearchResultImage: React.FC<{ imageUrl?: string | null }> = ({
+  imageUrl,
+}) => {
+  if (!imageUrl) return null;
+
+  return (
+    <div className="scene-image-container">
+      <img src={imageUrl} alt="" className="align-self-center scene-image" />
+    </div>
+  );
+};
+
+// Shared component for rendering tags
+const SearchResultTags: React.FC<{
+  tags?: GQL.ScrapedTag[] | null;
+}> = ({ tags }) => {
+  if (!tags || tags.length === 0) return null;
+
+  return (
+    <Row>
+      <Col>
+        {tags.map((tag) => (
+          <Badge className="tag-item" variant="secondary" key={tag.stored_id}>
+            {tag.name}
+          </Badge>
+        ))}
+      </Col>
+    </Row>
+  );
+};
+
+// Performer Result Component
+interface IPerformerResultProps {
+  performer: GQL.ScrapedPerformerDataFragment;
+}
+
+const PerformerSearchResultDetails: React.FC<IPerformerResultProps> = ({
+  performer,
+}) => {
+  const age = performer?.birthdate
+    ? TextUtils.age(performer.birthdate, performer.death_date)
+    : undefined;
+
+  return (
+    <div className="performer-result">
+      <Row>
+        <SearchResultImage imageUrl={performer.images?.[0]} />
+        <div className="col flex-column">
+          <h4 className="performer-name">
+            <span>{performer.name}</span>
+            {performer.disambiguation && (
+              <span className="performer-disambiguation">
+                {` (${performer.disambiguation})`}
+              </span>
+            )}
+          </h4>
+          <h5 className="performer-details">
+            {performer.gender && (
+              <span>
+                <GenderIcon
+                  className="gender-icon"
+                  gender={stringToGender(performer.gender, true)}
+                />
+              </span>
+            )}
+            {age && (
+              <span>
+                {`${age} `}
+                <FormattedMessage id="years_old" />
+              </span>
+            )}
+          </h5>
+          {performer.country && (
+            <span>
+              <CountryFlag
+                className="performer-result__country-flag"
+                country={performer.country}
+              />
+            </span>
+          )}
+        </div>
+      </Row>
+      <Row>
+        <Col>
+          <TruncatedText text={performer.details ?? ""} lineCount={3} />
+        </Col>
+      </Row>
+      <SearchResultTags tags={performer.tags} />
+    </div>
+  );
+};
+
+export const PerformerSearchResult: React.FC<IPerformerResultProps> = ({
+  performer,
+}) => {
+  return (
+    <div className="mt-3 search-item">
+      <PerformerSearchResultDetails performer={performer} />
+    </div>
+  );
+};
+
+// Studio Result Component
+interface IStudioResultProps {
+  studio: GQL.ScrapedStudioDataFragment;
+}
+
+const StudioSearchResultDetails: React.FC<IStudioResultProps> = ({
```

Comment:
I'm undecided on keeping the unrelated stuff in here, since we can't test it. Might be better to stash and remove it from this PR.